### PR TITLE
fix: item assets

### PR DESCRIPTION
**Pull request by jjfrench**
- Removes documentation and metadata links from item assets -> moves them into item links
- Removes duplicated properties, e.g boxes or title (was being set to item id)
- Adds method for custom asset roles based on file type, most useful for multi-data STAC records

*Also removed the entire cmr_json from being passed into functions

### CODE REVIEW

**Comment by jjfrench** [comment]
Before:
```
{
    "type": "Feature",
    "stac_version": "1.0.0",
    "id": "uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz",
    "properties": {
      "boxes": [
        "-2.0677778 9.1694444 0.61 11.8641667"
      ],
      "time_start": "2016-02-25T00:00:00.000Z",
      "updated": "2019-04-12T20:18:18.461Z",
      "dataset_id": "AfriSAR UAVSAR Vertical Wavenumber (KZ) Generated Using NISAR Tools",
      "data_center": "NASA_MAAP",
      "title": "uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz",
      "coordinate_system": "CARTESIAN",
      "day_night_flag": "UNSPECIFIED",
      "time_end": "2016-02-25T23:59:59.000Z",
      "original_format": "ECHO10",
      "browse_flag": false,
      "collection_concept_id": "C1200109243-NASA_MAAP",
      "online_access_flag": false,
      "links": [
        {
          "rel": "http://esipfed.org/ns/fedsearch/1.1/s3#",
          "title": "File to download",
          "hreflang": "en-US",
          "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.hdr"
        },
        {
          "inherited": true,
          "rel": "http://esipfed.org/ns/fedsearch/1.1/data#",
          "hreflang": "en-US",
          "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1"
        },
        {
          "inherited": true,
          "rel": "http://esipfed.org/ns/fedsearch/1.1/documentation#",
          "hreflang": "en-US",
          "href": "https://ieeexplore.ieee.org/document/8469014"
        }
      ],
      "concept_id": "G1200110083-NASA_MAAP",
      "datetime": "2016-02-25T00:00:00Z"
    },
    "geometry": {
      "coordinates": [
        [
          [
            -2.0677778,
            9.1694444
          ],
          [
            0.61,
            9.1694444
          ],
          [
            0.61,
            11.8641667
          ],
          [
            -2.0677778,
            11.8641667
          ],
          [
            -2.0677778,
            9.1694444
          ]
        ]
      ],
      "type": "Polygon"
    },
    "links": [
      {
        "rel": "self",
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.hdr",
        "type": "application/json"
      }
    ],
    "assets": {
      "bin": {
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.bin",
        "type": "binary/octet-stream",
        "roles": [
          "data"
        ]
      },
      "documentation": {
        "href": "https://ieeexplore.ieee.org/document/8469014",
        "roles": [
          "documentation"
        ]
      },
      "hdr": {
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.hdr",
        "type": "binary/octet-stream",
        "roles": [
          "data"
        ]
      },
      "vrt": {
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.vrt",
        "type": "binary/octet-stream",
        "roles": [
          "data"
        ]
      }
    },
    "bbox": [
      -2.0677778,
      9.1694444,
      0.61,
      11.8641667
    ],
    "stac_extensions": [],
    "collection": "AfriSAR_UAVSAR_KZ"
}
```
After:
```
{
    "type": "Feature",
    "stac_version": "1.0.0",
    "id": "uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz",
    "properties": {
      "time_start": "2016-02-25T00:00:00.000Z",
      "updated": "2019-04-12T20:18:18.461Z",
      "dataset_id": "AfriSAR UAVSAR Vertical Wavenumber (KZ) Generated Using NISAR Tools",
      "data_center": "NASA_MAAP",
      "coordinate_system": "CARTESIAN",
      "day_night_flag": "UNSPECIFIED",
      "time_end": "2016-02-25T23:59:59.000Z",
      "original_format": "ECHO10",
      "browse_flag": false,
      "collection_concept_id": "C1200109243-NASA_MAAP",
      "online_access_flag": false,
      "concept_id": "G1200110083-NASA_MAAP",
      "datetime": "2016-02-25T00:00:00Z"
    },
    "geometry": {
      "coordinates": [
        [
          [
            -2.0677778,
            9.1694444
          ],
          [
            0.61,
            9.1694444
          ],
          [
            0.61,
            11.8641667
          ],
          [
            -2.0677778,
            11.8641667
          ],
          [
            -2.0677778,
            9.1694444
          ]
        ]
      ],
      "type": "Polygon"
    },
    "links": [
      {
        "rel": "self",
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.hdr",
        "type": "application/json"
      },
      {
        "rel": "documentation",
        "href": "https://ieeexplore.ieee.org/document/8469014"
      }
    ],
    "assets": {
      "bin": {
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.bin",
        "type": "binary/octet-stream",
        "roles": [
          "data"
        ]
      },
      "hdr": {
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.hdr",
        "type": "binary/octet-stream",
        "roles": [
          "data"
        ]
      },
      "vrt": {
        "href": "s3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_KZ___1/uavsar_AfriSAR_v1-coreg_fine_lopenp_14043_16008_140_009_160225_kz.vrt",
        "type": "application/octet-stream",
        "roles": [
          "data"
        ]
      }
    },
    "bbox": [
      -2.0677778,
      9.1694444,
      0.61,
      11.8641667
    ],
    "stac_extensions": [],
    "collection": "AfriSAR_UAVSAR_KZ"
  }
```

**Comment by emileten** [comment]
Looks like we need to format and slightly modify the build-stac tests accordingly, can do that while reviewing @jjfrench.

File: lambdas/build-stac/utils/stac.py
Comment (abarciauskas-bgse) [inline lambdas/build-stac/utils/stac.py:256]: Looking at results for `prj file geospatial` in google led me to https://en.wikipedia.org/wiki/Shapefile - reading that page leads me to believe prj should actually be an asset, @wildintellect would you agree?

It seems like [prj files may be commonly used by ESRI applications](https://vsp.pnnl.gov/help/vsample/ESRI_PRJ_File.htm). And looking at that reference, I see that prj files are used to store metadata about a projection system.

**Comment by abarciauskas-bgse** [comment]
One minor comment, looks reasonable to me but also love the idea of adding tests per @emileten suggestion.

Thanks for doing this so quickly @jjfrench !

**Comment by emileten** [comment]
fixed/added a small test

**Comment by emileten** [comment]
LGTM (although there is a comment from aimee to address), fixed the tests.

**Review by emileten — APPROVED** [review]

File: lambdas/build-stac/utils/stac.py
Comment (jjfrench) [reply lambdas/build-stac/utils/stac.py:256]: Right, these would most likely be assets with metadata roles - guess I got a little carried away 😅

**Comment by jjfrench** [comment]
Tested on TEST stack using [ABoVE_UAVSAR_PALSAR](https://wm5rv6onbc.execute-api.us-west-2.amazonaws.com/collections/ABoVE_UAVSAR_PALSAR/items)

**Comment by emileten** [comment]
Checking the status of this, are we ready to merge @jjfrench ? Asking to know if I can use it to deploy my monday (https://github.com/NASA-IMPACT/active-maap-sprint/issues/488).

**Comment by jjfrench** [comment]
@emileten I was just waiting on approvals but yes, I believe this is ready to be merged

**Comment by jjfrench** [comment]
Merging for @emileten - can revert later if needed

**Pull request merged**
